pkg/messaging/rabbitmq/pubsub.go

```diff
@@ -79,7 +79,17 @@ func (ps *pubsub) Subscribe(id, topic string, handler messaging.MessageHandler)
 	case true:
 		// Check topic ID
 		if _, ok := s[id]; ok {
-			return ErrAlreadySubscribed
+			// Unlocking, so Unsubscribe() can access ps.subscriptions
+			ps.mu.Unlock()
+			if err := ps.Unsubscribe(id, topic); err != nil {
+				return err
+			}
+			ps.mu.Lock()
+
+			if len(s) == 0 {
```

Comment:
What is this? We are checking the length of a map within a loop i which we can go only if the element `id` exists (which should of course mean that map is not empty), and then in the code that is probably never reached we are setting subscriptions map to an uninitialized map `s`?

Reply:
@drasko Because map can be emptied by calling unsubscribe at L84. However, this is dangerous and will probably work only because maps are passed around by pointer. See my previous comment.